estimagic/optimization/optimize.py

```diff
@@ -418,3 +421,35 @@ def _process_results(res, params, internal_params, constraints, origin):
 
     res_dict["internal_x"] = x.tolist()
     return res_dict, params
+
+
+def _logging(params, fitness_eval, counter, log_interval=100, path_to_sql="logging.db"):
+    """Log parameter values and fitness.
+
+    Saves optimization results every `log_interval` iteration to `path_to_sql`.
+
+    Arg:
+        params (DataFrame): See :ref:`params`
+        fitness_eval (float): current fitness evaluation
+        counter (int): number of current iteration
+        log_interval (int): number of iterations until first logging shall
+                            occur
+        path_to_sql (str): path to sql database (needs to end with .db)
+
+    """
+    if counter % log_interval == 0:
+        # Reshaping Param DataFrame (Transposing)
+        p = params["value"].reset_index(drop=True).copy()
+        p_trans = pd.DataFrame(p).T
+        p_trans.columns = params["value"].index
+        p_final = p_trans.reset_index(drop=True)
+
+        # Save additional values
+        p_final["fitness_eval"] = fitness_eval
+        p_final["iteration"] = counter
+
+        # Save results as sql database
+        conn = sqlite3.connect(path_to_sql)
+        p_final.to_sql(name="params", con=conn, if_exists="append", index=False)
```

Comment:
If you append to existing databases, one would need to have a unique identifier for which run this is... 

Maybe create a new table each time with the start date at minute precision, overwrite if same date?

Reply:
so sth like if counter == 1, then get date, save table under data?

Reply:
I think you would just do that in the context manager, see above. 

I.e., in maximize/minimize:

1. Open the connection (better default name: `estimagic_optimization_log.sqlite`
2. Create the table `YYYY-MM-DD-HH-MM`
3. Pass the table and the connection as keyword arguments

Inside the internal criterion, pretty much as is, but only call it if a connection is around (`_log_params(table=kwargs['table'], conn=kwargs['conn'], params=params, fitness=f if kwargs['conn'] and true_iteration`).

I  like `_log_params` better than logging...

Reply:
true_iteration?

Reply:
`true_iteration` = iteration of the algorithm as opposed to function evaluation for calculating gradient and friends.